### PR TITLE
add a template for usage doc

### PR DESCRIPTION
**Pull request by mengqiy**

### CODE REVIEW

**Review by phanimarupaka — APPROVED** [review]

File: functions/template/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# some-function-name
+
+### Overview
+
+<!--mdtogo:Short-->
```
Comment (frankfarzan) [inline functions/template/README.md:5]: Is this applicable to only Go functions? If so, move it to functions/go?

What about TS functions? Are we rationalizing this with:

https://github.com/GoogleContainerTools/kpt-functions-sdk/blob/master/ts/create-kpt-functions/templates/func.mustache

Also, are usage docs for existing functions being changed to conform to this template?

Comment (mengqiy) [reply functions/template/README.md]: > Is this applicable to only Go functions? If so, move it to functions/go?

Moved it.

> What about TS functions?

GoogleContainerTools/kpt-functions-sdk#254

> Also, are usage docs for existing functions being changed to conform to this template?

Filed https://github.com/GoogleContainerTools/kpt/issues/1636

File: functions/go/template/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+Explain the function config and behavior for this function in detail. If the
+function supports both ConfigMap and CRD as function config, both should be
+documented.
+
+Note: The content between `<!--mdtogo:Long-->` and the following `<!--mdtogo-->`
+will be used as the long description for the command.
+
+<!--mdtogo-->
+
+### Examples
```
[the remark this answers is not in the document]
Comment (mengqiy) [reply functions/go/template/README.md:28]: TBH I don't have a super clear idea here.
IMO the example packages is a better way to present examples. The examples in the usage doc should be simpler, and it should look like what we usually in a `quick start` doc.

@frankfarzan WDYT?

[the remark this answers is not in the document]
Comment (mengqiy) [reply functions/go/template/README.md:29]: After thinking more about it, let's put a link to the examples web pages here.

File: functions/go/template/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+`<!--mdtogo-->` will be used as the long description for the command.)
+
+<!--mdtogo-->
+
+### Examples
+
+<!--mdtogo:Examples-->
+
+Add a link to the web page of the examples. Examples should be added in the
+`examples` directory.
```
Comment (frankfarzan) [inline functions/go/template/README.md:34]: What `examples` directory? Does this README apply to any function or only functions in the catalog where we have this directory convention?

File: functions/go/template/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+Explain what this function does in one or two sentences.
+
+Note: The content between `<!--mdtogo:Short-->` and the following
+`<!--mdtogo-->` will be used as the short description for the command.
+
+<!--mdtogo-->
+
+
+### Synopsis
```
[the remark this answers is not in the document]
Comment (mengqiy) [reply functions/go/template/README.md:15]: Declarative flow also depends on a Kptfile besides Kpt CLI.
IMO we don't need to show kpt CLI and Kptfile, we can do that in the examples.
In this section, we should focus on explaining the fn config and the function's behavior.

File: functions/go/template/README.md (hunk 3 above)
Comment (mengqiy) [reply functions/go/template/README.md]: Updated.

File: functions/go/template/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (frankfarzan) [reply functions/go/template/README.md:15]: setter and search and replace usage docs do that today. @phanimarupaka

[the remark this answers is not in the document]
Comment (phanimarupaka) [reply functions/go/template/README.md:15]: Ack.

[the remark this answers is not in the document]
Comment (phanimarupaka) [reply functions/go/template/README.md:15]: @frankfarzan https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/201

**Review by frankfarzan — APPROVED** [review]

**Pull request merged**